02-express-tutorial/controllers/products.js

```diff
@@ -0,0 +1,48 @@
+let { products } = require("../data");
+
+const getProducts = (req, res) => {
+  res.status(200).json({ success: true, data: products });
+};
+
+const getProduct = (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find((p) => p.id === idToFind);
+
+  if (!product) {
+    res.status(404).json({ message: "No such a product!" });
+  } else {
+    res.status(200).json(product);
+  }
```

Comment:
Nitpick (doesn't require changes):
This can be simplified using the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator).
```js
return product
  ?  res.status(200).json(product);
  :  res.status(404).json({ message: "No such a product!" });
```